icicle/include/icicle/hash/hash.h

```diff
@@ -22,7 +22,7 @@ namespace icicle {
      *
      * @param backend A shared pointer to a backend hash implementation.
      */
-    explicit Hash(std::shared_ptr<HashBackend> backend) : m_backend(std::move(backend)) {}
+    explicit Hash(std::shared_ptr<HashBackend> backend = nullptr) : m_backend(std::move(backend)) {}
```

Comment:
maybe better let the caller pass nullptr rather than default value so that this is explicit and not implicit.

Reply:
done